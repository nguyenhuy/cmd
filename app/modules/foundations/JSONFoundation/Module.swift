Target.module(
  name: "JSONFoundation",
  dependencies: [],
  testDependencies: [])
