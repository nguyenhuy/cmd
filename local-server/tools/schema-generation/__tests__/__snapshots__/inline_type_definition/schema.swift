// swiftformat:disable all
// This file is generated from ./local-server/src/server/schemas/schemaFile.ts by `yarn export-schema-swift`.
// Do not edit it manually.

import Foundation
import JSONFoundation

extension Schema {
  public struct ValueType: Codable, Sendable {
    public let nested: Nested
  
    private enum CodingKeys: String, CodingKey {
      case nested = "nested"
    }
  
    public init(
        nested: Nested
    ) {
      self.nested = nested
    }
  
    public init(from decoder: Decoder) throws {
      let container = try decoder.container(keyedBy: CodingKeys.self)
      nested = try container.decode(Nested.self, forKey: .nested)
    }
  
    public func encode(to encoder: Encoder) throws {
      var container = encoder.container(keyedBy: CodingKeys.self)
      try container.encode(nested, forKey: .nested)
    }
  
    public struct Nested: Codable, Sendable {
      public let value: String
    
      private enum CodingKeys: String, CodingKey {
        case value = "value"
      }
    
      public init(
          value: String
      ) {
        self.value = value
      }
    
      public init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: CodingKeys.self)
        value = try container.decode(String.self, forKey: .value)
      }
    
      public func encode(to encoder: Encoder) throws {
        var container = encoder.container(keyedBy: CodingKeys.self)
        try container.encode(value, forKey: .value)
      }
    }
  }}
