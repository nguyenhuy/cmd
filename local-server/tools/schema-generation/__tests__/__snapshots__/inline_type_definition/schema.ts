export interface ValueType {
  nested: {
    value: string;
  };
}
