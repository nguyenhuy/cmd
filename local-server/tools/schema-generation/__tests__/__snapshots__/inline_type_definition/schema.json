{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "$ref": "#/definitions/ValueType",
  "definitions": {
    "ValueType": {
      "type": "object",
      "properties": {
        "nested": {
          "type": "object",
          "properties": {
            "value": {
              "type": "string"
            }
          },
          "required": [
            "value"
          ],
          "additionalProperties": false
        }
      },
      "required": [
        "nested"
      ],
      "additionalProperties": false
    }
  }
}